ruby/basic_ruby/predicate_enumerable_methods.md

```diff
@@ -164,7 +164,7 @@ fruits.all? { |fruit| fruit.length > 6 }
 Special note to keep in mind while debugging: `#all?` will return `true` by default unless the block returns `false` or `nil`. So if you call `#all?` on an empty array or hash (i.e., there are no elements  for the block to evaluate), it will return `true`.
 
 ### The none? method
-As you might expect, `#none?` performs the opposite function of `#all?`. It returns `true` only if the condition in the block matches *none* of the elements in your array or hash; otherwise, it returns `false`.
+As you might expect, `#none?` performs the opposite function of `#any?`. It returns `true` only if the condition in the block matches *none* of the elements in your array or hash; otherwise, it returns `false`.
```

Comment:
Thanks for taking the time to contribute @dimbyr. But I don't think this is correct, functionally `none` is closer to being the opposite of `all` in ruby.

To put them another way:
`#all?` - returns true if all of the elements match the condition in the block
`#none?` - returns true if **all** the elements do not match the condition in the block

Reply:
@KevinMulhern 

The negation  of `#all`, which I refer to as `not #all`,  returns true if **some** of the elements **do not** satisfy the condition in the block. However  it is possible that some of them **do satisfy** the condition. 

The following points show that `#any` is the opposit of `#none`, i.e, these two methods can never return the same value for the same array.  

1. If `#none` returns `false`, this means that there is at least one element that satisfy the property. In this case, `#any` returns `true`.  
2. One can also see that if `#any` returns `true`, then `#none` will return `false`.
3. If `#none` returns `true`, this means that there is no element satisfying the property. In this case, `#any` returns `false`. 
4. Similarly, if `#any` returns `true`, then `#none` will return `false`.

Reply:
I see both sides and wonder if we could clear up the confusion by just getting rid of the "opposite" explanation. 
```suggestion
As you might expect, `#none?` returns `true` only if the condition in the block matches *none* of the elements in your array or hash; otherwise, it returns `false`.
```

Reply:
Thanks for the detailed response @dimbyr, I can see where you're coming from better now. 

Semantically both any and all are the opposite of none so I think what @rlmoser99 is suggesting is the best way forward.

Reply:
@KevinMulhern ,

Sorry, I wanted to clarify that `#none?` is NOT the opposite of `#all?`. 

Of course `#none?` returns `false` when `#all?` returns `true`, BUT one does not need `#all?` to return `true` for `#none?` to return `false`.

So, the two sentences on `line 167` is correct IF AND ONLY IF one replace `#all?` with `#any?`

Thanks.

Reply:
<img width="834" alt="Screenshot 2023-08-30 at 06 58 43" src="https://github.com/TheOdinProject/curriculum/assets/7963776/1b90978e-dc27-43ee-a8ee-ba9a19c105b0">

Reply:
I think w might be splitting hairs at this point, both all and any are the opposite of none **semantically**. To get this PR merged can we remove the opposite comparison please?

Reply:
Yes, I agree, they are semantically opposite, but in our context, the two methods do not always perform actions that are opposite to each other. As I have mentioned, they can return `false` at the same time.

Reply:
I think the word "some"  in the list of opposites in the dictionary is the one that translates better the opposite of `#all?` in our context.

Reply:
@KevinMulhern,
Sorry, maybe I was not clear enough. 
My point is that the statement " `#none?` performs the opposite function of `#all?`. " is not correct. I do not mean the words "all" and "none" are not semantically opposite.

Reply:
Thats a fair point and I appreciate where you're coming from. I think we've maybe got a different view on what the context of this is. You're coming at it from a mathematics standpoint and I'm coming at it from what intention you'll want to communicate through the code.

If we both disagree, then theres a good chance students will be confused with this too. I think doing as Rlmoser suggested by removing this opposite comparison entirely is the best compromise and course of action.